File: contains_duplicate/contains_duplicate.java
import java.util.*;

class Solution {
    private static Set<Integer> duplicateList(int[] nums) {
        Set<Integer> seen = new HashSet<Integer>();
        Set<Integer> seen1 = new HashSet<Integer>();
        for (int n : nums) {
            if (seen.contains(n)) {
                seen1.add(n);
            } else {
                seen.add(n);
            }
        }
        return seen1;
    }
    
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = duplicateList(nums);
        if (set.size() > 0) {
            return true;
        } else {
            return false; 
        }
    }
}

File: contains_duplicate/contains_duplicate.py
class Solution(object):
    def duplicateList(self, nums):
        seen = set()
        seen2 = set()
        seen_add = seen.add
        seen2_add = seen2.add
        for item in nums:
            if item in seen:
                seen2_add(item)
            else:
                seen_add(item)
        return list(seen2)
    
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        nums_set = Solution().duplicateList(nums)
        if len(nums_set) > 0:
            return True
        else:
            return False
